packages/composites/historical-average/src/dataProvider.ts

```diff
@@ -0,0 +1,61 @@
+import { RequestConfig, AdapterResponse } from '@chainlink/types'
+import { Logger, Requester } from '@chainlink/ea-bootstrap'
+import * as cmc from '@chainlink/coinmarketcap-adapter'
+
+export type ResponsePayload = {
+  timestamp: Date
+  price: number
+}[]
+
+export const getPriceProvider =
+  (source: string, jobRunID: string, apiConfig: RequestConfig) =>
+  async (
+    base: string,
+    quote: string,
+    fromDate: Date,
+    toDate: Date,
+    interval: string,
+  ): Promise<ResponsePayload> => {
+    try {
+      switch (source.toUpperCase()) {
+        case cmc.NAME:
+          return getCoinMarketCapPrice(jobRunID, base, quote, fromDate, toDate, interval, apiConfig)
+      }
```

Comment:
Can you add a `default` here?

Reply:
I purposefully didn't put it here because this is happening inside a try catch, so throwing an error in default here would just make it get caught and wrapped in a misleading error.

Do you think it's worth refactoring to support `default` here?

Reply:
Oh, I see the `throw new Error` on line 31. Yeah it's a little funny but I think it's fine. IMO not worth refactoring 👍